src/lib/Renderer.ts

```diff
@@ -353,7 +343,127 @@ class Renderer {
       return { error: 'unsafe_redirect' };
     }
 
-    return { statusCode, headers, body, timeout, resolvedUrl };
+    return { statusCode, headers, body, resolvedUrl };
+  }
+
+  private async _processLogin(task: TaskParams): Promise<TaskResult> {
+    /* Setup */
+    const { url } = task;
+    const { context, page } = await this._newPageWithContext(task);
+
+    try {
+      await this._goto(page, url);
+    } catch (e) {
+      return { error: e.message, timeout: Boolean(e.timeout) };
+    }
+
+    const textInput = await page.$('input[type=text], input[type=email]');
+    if (!textInput) {
+      return { error: `field_not_found: input[type=text], input[type=email]` };
+    }
+    await textInput!.type(task.login!.username);
+
+    let passwordInput = await page.$('input[type=password]');
+    if (!passwordInput) {
+      console.log('2 step login: validating username...');
+      try {
+        await Promise.all([
+          // page.waitForNavigation(), // Doesn't work with Okta for example, it's JS based
+          page.waitForSelector('input[type=password]', {
+            timeout: TIMEOUT,
+          }),
+          textInput!.press('Enter'),
+        ]);
+      } catch (err) {
+        console.log('Found no password input on the page');
+        const body = await this._renderBody(page, new URL(page.url()));
+        return { error: err.message, body };
+      }
+    }
+
+    console.log(`2 step login: navigated to ${page.url()}`);
+    passwordInput = await page.$('input[type=password]');
+    console.log('Logging in...');
+    await passwordInput!.type(task.login!.password);
+    let loginResponse;
+    try {
+      const [navigationResponse] = await Promise.all([
+        page.waitForNavigation({ timeout: TIMEOUT }),
+        await passwordInput!.press('Enter'),
```

Comment:
There is a potential issue here, if the form redirects to our internal API or a foreign domain.
Might be complicated to solve

Reply:
I'd say that both the URL of the form and the credentials are given by users, so it's up to their website login to be secure.

Reply:
In case of honest user yes, in case of hacker you setup a form that redirects to an "internal API" and you get our secrets. Highly unlikely but doable.  
I'd say, we need to pass allowedDomain as param, intercepts redirection, validate  and document that SSO should be added to allowed domain validation.

We tried something with, but it's already too late, if for example the call was `DELETE /important_resource`
```js
    if (preSerializationUrl !== resolvedUrl) {
      // something super shady happened where the page url changed during evaluation
      return { error: 'unsafe_redirect' };
    }
```

Reply:
Oh I see, indeed that's an issue.

Reply:
we can think of something in an other PR to validate this one

Reply:
Created a ticket